cpp/test/writer/time_chunk_writer_test.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License a
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "writer/time_chunk_writer.h"
+
+#include <gtest/gtest.h>
+
+#include "common/allocator/byte_stream.h"
+#include "common/config/config.h"
+#include "common/statistic.h"
+
+using namespace storage;
+using namespace common;
+
+class TimeChunkWriterTest : public ::testing::Test {
+   protected:
+    TimeChunkWriter time_chunk_writer;
+    ColumnDesc col_desc;
+
+    void SetUp() override {
+        col_desc.column_name_ = "time";
+        col_desc.type_ = TSDataType::VECTOR;
+        col_desc.encoding_ = TSEncoding::PLAIN;
+        col_desc.compression_ = CompressionType::UNCOMPRESSED;
+
+        ASSERT_EQ(time_chunk_writer.init(col_desc), E_OK);
+    }
+
+    void TearDown() override { time_chunk_writer.destroy(); }
+};
+
+TEST_F(TimeChunkWriterTest, InitWithColumnDesc) {
+    EXPECT_EQ(time_chunk_writer.init(col_desc), E_OK);
+}
+
+TEST_F(TimeChunkWriterTest, InitWithParameters) {
+    TimeChunkWriter writer;
+    EXPECT_EQ(writer.init("time", TSDataType::DOUBLE, TSEncoding::PLAIN,
+                          CompressionType::UNCOMPRESSED),
+              E_OK);
+    writer.destroy();
+}
```

Comment:
DOUBLE should not be a legal parameter for TimeChunk.
Actually, I do not think TimeChunk needs a data type as a parameter at all.

Reply:
Removed the redundant data type parameters.